include/round.h

```diff
@@ -15,427 +15,559 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-/*
-    This file complements stdfix-full-iso.h to add various rounding routines
-    and mixed-format fixed-point multiplication routines.
-
-    Note that GCC currenly does not implement rounding on such operations like
-    fixed-point multiplication and conversion (e.g. decimal to fixed-point).
-    Some more background in https://arxiv.org/abs/1904.11263
-
-    Author: mantas.mikaitis@manchester.ac.uk, 2019 August.
-
+/*! \file
+ * \brief Fixed-point multiplication with controlled rounding
+ *
+ * This file complements stdfix-full-iso.h to add various rounding routines
+ * and mixed-format fixed-point multiplication routines.
+ *
+ * Note that GCC currently does not implement rounding on such operations like
+ * fixed-point multiplication and conversion (e.g. decimal to fixed-point).
+ * Some more background in https://arxiv.org/abs/1904.11263
+ *
+ * \author mantas.mikaitis@manchester.ac.uk
+ * \date 2019 August.
  */
 
+#ifndef __ROUND_H__
+#define __ROUND_H__
+
 #include <stdfix-full-iso.h>
 #include <random.h>
 #include <sark.h>
 #include "utils.h"
 
 #include <stdfix.h>
 
-#ifndef _ROUND_
-#define _ROUND_
-
-// Choose random number generator that is used for the multiplications with
-// stochastic rounding
-//#define RANDOM() mars_kiss64_simp()
-//#define RANDOM() mars_kiss32()
+//! Choose random number generator that is used for the multiplications with
+//! stochastic rounding
+#if 0
+#define RANDOM() mars_kiss64_simp()
+#elif 0
+#define RANDOM() mars_kiss32()
+#elif 1
```

Comment:
Yes. Before I touched the code, switching was by uncommenting one of the cases and making the others commented out. This improved version at least gives a better hint that these are at least pseudo-live code paths rather than totally dead.

I'm not solving this problem in this PR.